tests-e2e/ocw-ci-test-course/headings.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { test, expect } from "@playwright/test"
+import { CoursePage } from "../util"
+
+test(`Verify accesibility and semantic structure of headings`, async ({
+  page
+}) => {
+  const headingsData = [
+    {
+      label:                "OCW CI Test Course",
+      expectedHeadingLevel: "H1"
+    },
+    {
+      label:                "Section 2",
+      expectedHeadingLevel: "H2"
+    },
+    {
+      label:                "H3 heading",
+      expectedHeadingLevel: "H3"
+    },
+    {
+      label:                "Course Info",
+      expectedHeadingLevel: "H4"
+    },
+    {
+      label:                "INSTRUCTORS",
+      expectedHeadingLevel: "H5"
+    },
+    {
+      label:                "DEPARTMENTS",
+      expectedHeadingLevel: "H5"
+    },
+    {
+      label:                "AS TAUGHT IN",
+      expectedHeadingLevel: "H5"
+    },
+    {
+      label:                "LEVEL",
+      expectedHeadingLevel: "H5"
+    },
+    {
+      label:                "TOPICS",
+      expectedHeadingLevel: "H5"
+    },
+    {
+      label:                "LEARNING RESOURCE TYPES",
+      expectedHeadingLevel: "H5"
+    }
+  ]
+  const coursePage = new CoursePage(page, "course")
+  await coursePage.goto("pages/assignments")
+
+  for (const { label, expectedHeadingLevel } of headingsData) {
+    const heading = page.getByRole("heading", { name: label })
+    expect(heading).not.toBeNull()
```

Comment:
**Comment:** I'm fairly certain that this expectation does not do anything: On line 53, `heading` is a [locator](https://playwright.dev/docs/api/class-locator). It is absolutely not null. No doubt about it.

In particular, the locator has not yet resolved to any particular HTML elements. If you want to be assured that it matches before moving on, you can call `await heading.waitFor()`.

Reply:
Thanks. Updated